### ISSUE TITLE
Perceptual Loss missing

### ISSUE THREAD

**Issue by Cuky88** [taki0112/MUNIT-Tensorflow#9]
Thanks for your great work @taki0112 

I'm curious why you didn't implement the perceptual loss, is there a special reason?

Cheers.

**Comment by taki0112** [comment]
To make the code more simpler
In original [MUNIT](https://github.com/NVlabs/MUNIT), use the [pretrained_vgg16_lua_version](https://github.com/abhiskk/fast-neural-style/blob/master/neural_style/utils.py#L65)

So to load this model, we need the `load_lua` function in `pytorch`
I want to write the code only using tensorflow, so I did't implement the perceptual loss

However, I will also make it possible to do with tensorflow.

Thanks

**Comment by Cuky88** [comment]
Hi, thanks for the fast reply.

I saw that and I'm already working on a solution in tf only.

**Comment by taki0112** [comment]
Is it possible to PR?

**Comment by Cuky88** [comment]
Yes, I'll send you a PR when I'm finished

**Comment by taki0112** [comment]
Thanks a lot.

**Comment by Cuky88** [comment]
@taki0112 Hi, I think I managed to get perceptual loss to work, but I'm not 100% sure.

Unfortunately I cannot create a PR, since I changed a couple of other things before. Please look at this [commit](https://github.com/Cuky88/MUNIT-Tensorflow/commit/9275830667cda0c052a4afcb02040c4e5a1fed72). There is everything for perceptual loss. You can find the [download links](https://github.com/Cuky88/MUNIT-Tensorflow/blob/master/vgg16.py#L15) for the vgg16 weight files in vgg16.py on top in comment section. You can also look in [this config file](https://github.com/Cuky88/MUNIT-Tensorflow/blob/master/configs/sun2rain_perl.yaml#L31) to see which values the arguments for the vgg part has. I only tested everything with the .h5 weight files.

Another issue would be to implement [LPIPS Distance](https://github.com/richzhang/PerceptualSimilarity) also in TF. Do you have such intentions?

It would be very good if you could look over the code, I'm not an expert in TF like you :)

EDIT: [this commit](https://github.com/Cuky88/MUNIT-Tensorflow/commit/aa0b2618a87c51b42ed0aceda9897d4aaa7af392) is also needed.

**Comment by MartinMeliss** [comment]
@Cuky88 Hi.
I think that you have a bug in the "vgg_preprocess" method in "ops.py".
You subtract means from the image, but the range of the image is -1: 1, and your values for the byte representation.
And, as far as I understand, VGG-16 requires an unregulated input:
line #263 ops.py:
`channels[i] -= means[i]`
should look like:
`channels[i] = (channels[i] + 1.0) * 127.5 - means[i]`

Сorrect me if I'm wrong.

**Comment by Cuky88** [comment]
@MartinMeliss You are right, image floats are scaled between -1 and 1. So the vgg preprocessing is skrewing up everything. Thanks for pointing out, will fix soon.
